### PR TITLE
fix(Core/Gameobject): Spell spawned chests should not despawn on loot

### PR DESCRIPTION
**Pull request by Zoidwaffle**
## Changes Proposed:
-  Don't despawn temporary quest-chests when looted once
- This was taken from [TC](https://github.com/TrinityCore/TrinityCore/pull/22124/files)

## Issues Addressed:
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/4403
- There should be many more quests with the same issue... See other references from [TC](https://github.com/TrinityCore/TrinityCore/pull/22124)

## Tests Performed:
- Compiled the core without errors
- Looted the phylactery from Araj in different ways, it seems to work as intended

## How to Test the Changes:

This is for a single quest only!
- .go c id 1852
- .quest add 105 (Horde - or 211 for Alliance)
- /tar araj
- .die
- Loot the phylactery... it despawns right after looting unless the patch is applied

## Known Issues and TODO List:

- [ ] I do not claim to fully understand if these changes have unintentional side effects... Please help testing or sanity-checking if you can!

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Review by Helias — APPROVED** [review]

File: src/server/game/Entities/GameObject/GameObject.cpp [hunk 1]
```diff
@@ -706,9 +706,18 @@ void GameObject::Update(uint32 diff)
...
+                        {
+                                UpdateObjectVisibility();
+                                SetLootState(GO_READY);
+                        }
+                        else
+                        {
+                                SetRespawnTime(0);
+                                Delete();
+                        }
+                        return;
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/GameObject/GameObject.cpp:720]: isn't a bit too much space for indentation? we usually use 4 spaces

Comment (Zoidwaffle) [reply src/server/game/Entities/GameObject/GameObject.cpp]: Fair enough, I just lined up as with tabs... I changed it to 4 spaces now :smile:

**Comment by pklloveyou** [comment]
```
.ad 32566
```
Can this be loot all the time?

**Comment by Zoidwaffle** [comment]
Apparently, yes... The umbrella seems to despawn, but the picnic grill can be looted as many times as you want.... It stayed there for a few minutes or so before disappearing.

So... Is that good or bad? :eyes:

**Comment by pklloveyou** [comment]
It should disappear immediately after the loot

**Comment by pklloveyou** [comment]
I remember AZ restoring the code a long time ago

**Comment by Zoidwaffle** [comment]
Anyway, the picnic grill should be lootable as far as I can see here https://www.youtube.com/watch?v=KA0_zE_-OZk - so that's actually an improvement I believe.

So far it seems like this patch does fix several items - but does it break stuff? Well, I did also leave a warning in the pull request, I have no idea if it breaks more than it fixes :grin:

**Comment by Kitzunu** [comment]
From code logic this shouldn't break anything.

I think @pklloveyou is referring to this PR https://github.com/azerothcore/azerothcore-wotlk/pull/2750 which was reverted

**Comment by pklloveyou** [comment]
YES

**Comment by pklloveyou** [comment]
If only we knew all the affected spells, we could test them all

**Comment by Footman** [comment]
This pull request is not enough. There needs to be done more thing that depend on restockTime from DBC. Because right now you dont refill loot into chest after it was hidden here. TC did good job there already

**Comment by claudiodfc** [comment]
### Tested Sucessfully!

1. Killed Araj while having quest 211 as an Alliance character
2. Looted the Araj's Phylactery Shard item
3. Gameobject doesn't despawn after being looted

image

**Pull request merged**